SimulationAgent/Agent.cs

```diff
@@ -59,6 +65,16 @@ public async Task RunAsync()
             while (this.running)
             {
                 var oldSimulation = this.simulation;
+                DateTime now = DateTime.Now;
+                TimeSpan duration = now - this.lastPolledTime;
+                
+                // Send heartbeat every 24 hours
+                if (duration.Days >= DIAGNOSTICS_POLLING_FREQUENCY)
+                {
+                    this.lastPolledTime = DateTime.Now;
+                    this.logDiagnostics.LogServiceHeartbeatAsync();
+                }
```

Comment:
pls move all the heartbeat code into a separate private method, and call it from here. e.g. `await this.SendSolutionHeartbeatAsync()`

you could also consider moving the logic into the DiagnosticsLogger instead of having it here